src/types/ledger/inclusion_state.rs

```diff
@@ -18,6 +21,22 @@ pub enum LedgerInclusionState {
     NoTransaction,
 }
 
+impl Display for LedgerInclusionState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            LedgerInclusionState::Conflicting => write!(f, "conflicting"),
+            LedgerInclusionState::Included => write!(f, "included"),
+            LedgerInclusionState::NoTransaction => write!(f, "noTransaction"),
+        }
+    }
+}
+
+impl From<LedgerInclusionState> for Bson {
+    fn from(val: LedgerInclusionState) -> Self {
+        val.to_string().into()
+    }
+}
+
```

Comment:
The `Display` implementation is error prone, because it could get out of sync with the `#[serde(rename = "...")]`. Maybe it's better to do the following:

```suggestion
impl From<LedgerInclusionState> for Bson {
    fn from(val: LedgerInclusionState) -> Self {
        // Panic: This can never fail because we implement `Deserialize` on the type.
        mongodb::bson::to_bson(&val).unwrap()
    }
}

```